[Calvin Tan] iP

calvintanwj opened the pull request:
# Duke
> "If debugging is the process of removing software bugs, then programming must be the process of putting them in" - Edsger Dijkstra

Get your life together with Duke. It's

* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. Download the jar from [here](https://github.com/calvintanwj/ip/releases/tag/A-Jar)
2. Run the jar in your CLI.
3. Add and manage tasks to your heart's desire.  😉

And most importantly, it's  __FREE!__

Features implemented:

 - [X] Managing tasks
 - [X] Managing deadlines
 - [X] Reminders 
 - [ ] GUI (coming soon)

#

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

calvintanwj committed:
Implement greet, echo, and exit

calvintanwj committed:
Implement add, list functionality

calvintanwj committed:
Add ability to mark tasks as done

calvintanwj committed:
Add types of tasks (ToDos, Deadlines, Events)

calvintanwj committed:
Semi-automate testing

calvintanwj committed:
Add simple exception handling

calvintanwj committed:
Add delete functionality

calvintanwj committed:
Add more tests and exception handling

calvintanwj committed:
Refactor code

calvintanwj committed:
Add storage for task lists

After program exits, list of tasks is not saved.

By storing the list of tasks in a text file locally on your computer.
Even after the program exits, the bot can recreate the task list from
the text file, when users interact with the bot again.

This is more sensible, as users will not have their task lists wiped,
and will not have to recreate the list again.

calvintanwj committed:
Add dates and times to chatbot

Chatbot saves the timing of certain tasks as strings.

The timings are now saved as LocalDate objects.

This allows for more flexibility and ease when we implement future
commands such as showing tasks that occur within a given time period, as
the class LocalDate has a lot of its own built-in fuctionalities.

calvintanwj committed:
Merge branch 'branch-Level-7'

calvintanwj committed:
Merge branch 'branch-Level-8'

calvintanwj committed:
Refactor code

calvintanwj committed:
Make the code more OOP

Main logic was inside the Duke.java.

Closely related code has been extracted out into several classes such as
Command, Ui, TaskList, Parser,...

This allows for cleaner and more readable code, as the Duke.java file
was getting too clustered. Reduces boilerplate code, as well as makes
future implementations easier.

calvintanwj committed:
Divide classes into packages

Classes all stored into one folder.

Packaging related classes into packages, helps to prevent naming
conflicts, and makes searching for classes easier and cleaner.

calvintanwj committed:
Touch up Code

calvintanwj committed:
Add JUnit tests

No testing was done currently.

By adding JUnit tests, it allows us to easily add tests in the future,
as well as automate our testing, ensuring public methods/classes are
working as intended.

calvintanwj committed:
Add manifest file for building artifact

calvintanwj committed:
Add find command

Currently, users are unable to find a set of tasks in their lengthy task
lists.

Now, by implementing a new command, FindCommand, users can then search
for a specified keyword, and in return receive a set of tasks that match
the keyword.

calvintanwj committed:
Merge branch 'branch-Level-9'

calvintanwj committed:
Tweak code to comply with coding standard.

Code initially was disorganized and did not follow a set coding
standard.

Following a coding standard allows for better collaboration between
peers, and increases readability of your code.

calvintanwj committed:
Merge branch 'branch-A-CodingStandard'

calvintanwj committed:
Add JavaDoc comments

Allows for your code to be readable by anyone and future you.

calvintanwj committed:
Merge branch 'branch-A-JavaDoc'

calvintanwj committed:
Merge branch 'add-gradle-support'

calvintanwj committed:
Automate project build using Gradle

Previously, we would have to re-compile and re-run our programs
manually.

Gradle automates the build processes.

calvintanwj committed:
Add checkstyle

Use checkstyle to detect coding style violations.

calvintanwj committed:
Add a GUI to Duke.

calvintanwj committed:
Touch up UI and Fix bugs

DesmondTo commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,78 @@
+    private ImageView displayPicture;
+
+    /**
+     * The constructor for the Dialog Box.
```
I believe there should have empty line between description and parameter section 👍

DesmondTo commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    /**
+     * Gives the user dialog box.
```
I believe there should have empty line between description and parameter section 👍

DesmondTo commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    /**
+     * Flips content and gives the Duke bot dialog box.
```
I believe there should have empty line between description and parameter section 👍

DesmondTo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,85 @@
+import duke.ui.Ui;
+
+/**
+ * The Duke program implements a bot with a set of simple commands
```
I think you miss a "fullstop" at the end of this header comment 👍

DesmondTo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    /**
+     * Load tasks, returning the appropriate message.
```
I believe there should have empty line between description and parameter section 👍

DesmondTo commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    /**
+     * Executes the main logic of the bye command.
```
I believe there should have empty line between description and parameter section 👍

DesmondTo commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    /**
+     * Checks whether command terminates the program.
```
I believe there should have empty line between description and parameter section 👍

DesmondTo commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    /**
+     * Executes the main logic of the command
```
I think you miss a "full stop" here 👍

DesmondTo commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,54 @@
+     * @param tasks   The user's list of tasks.
+     * @param ui      The ui interacting with the user.
+     * @param storage The location where the list of tasks is stored.
+     * @throws DukeException
```
I think you miss a "full stop" here 👍

DesmondTo commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,72 @@
+public class DeadlineCommand extends Command {
+
+    /**
+     * The arguments associated with the command
```
I think you miss a "full stop" here 👍

DesmondTo commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,72 @@
+     * Executes the main logic of the command.
+     *
+     * @param tasks   The user's list of tasks.
+     * @param ui      The ui interacting with the user.
```
Although it looks neat to me, but I am not sure if we need to align the parameter in this way.

DesmondTo commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,83 @@
+        default:
+            throw new DukeException("I'm sorry, but I don't know what \""
+                    + fullCommand + "\" means :-(");
+
```
I think it is good to remove this redundant line here 👍

DesmondTo commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+     * Constructs a deadline.
+     *
+     * @param description The description associated with the deadline.
+     * @param by          The date by which the deadline should be completed.
```
Looks neat to me but not sure if need to align the parameter in this way 👍

DesmondTo commented on src/test/java/duke/command/DeleteCommandTest.java:
```diff
@@ -0,0 +1,56 @@
+        assertEquals(tasks.size(), 0);
+    }
+
+    // Can also handle if index is not an integer
```
Not sure if it is intended comment, but if not, may be it's a good idea to remove it 👍

DesmondTo reviewed (commented):
The code itself follows most of the coding standard, well done! There are just some nits to fix on your header comments. :)

calvintanwj committed:
Fix GUI

calvintanwj committed:
Add some assertions to Duke

Code previously did not make any use of assertions.

Added simple assertions to Duke to ensure the program is running
correctly by verifying certain conditions.

Using assertions increases confidence in the correctness of the program.

calvintanwj committed:
Merge pull request #1 from calvintanwj/branch-A-Assertions

Add some assertions to Duke

calvintanwj committed:
Improve code quality

A lot of duplicated code was seen amongst commands.

Moved a lot of duplicated code into parser class. Removed dead code
related to the old CLI interface.

Refactoring helps to make it easier for future extensions to the system.

calvintanwj committed:
Merge pull request #2 from calvintanwj/branch-A-CodeQuality

Improve code quality

calvintanwj committed:
Use Streams

calvintanwj committed:
Add CI

luffingluffy commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    /**
+     * Boot up duke and load up task list.
```
I think you can change this to 'boots' instead.

luffingluffy commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,56 @@
+public class FindCommand extends Command {
+
+    /**
+     * The arguments associated with the command
```
I think you missed a full stop here.

luffingluffy reviewed (commented):
Some minor nitpickings but LGTM otherwise!

calvintanwj committed:
Touch up doc headers

calvintanwj commented:
Thank you both for your feedback!

calvintanwj committed:
Add shorter aliases for commands

calvintanwj committed:
Merge branch 'branch-C-FriendlierSyntax'

calvintanwj committed:
Improve GUI

calvintanwj committed:
Update README.md

calvintanwj committed:
Add Ui screenshot

calvintanwj committed:
Update README.md
